cg/services/orders/order_status_service/utils.py

```diff
@@ -0,0 +1,87 @@
+from cg.apps.tb.dto.summary_response import AnalysisSummary
+from cg.services.orders.order_status_service.dto.order_status_summary import OrderSummary
+from cg.services.orders.order_status_service.dto.case_status_summary import CaseSummary
+from cg.store.models import Case, Order
+
+
+def create_status_summaries(orders: list[Order]) -> list[OrderSummary]:
+    summaries: list[OrderSummary] = []
+    for order in orders:
+        case_count: int = get_total_cases_in_order(order)
+        summary = OrderSummary(order_id=order.id, total=case_count)
+        summaries.append(summary)
+    return summaries
+
+
+def add_analysis_summaries(
+    order_summaries: list[OrderSummary],
+    analysis_summaries: list[AnalysisSummary],
+) -> None:
+    order_summary_map = {summary.order_id: summary for summary in order_summaries}
+    for analysis_summary in analysis_summaries:
+        order_summary = order_summary_map[analysis_summary.order_id]
+        order_summary.delivered = analysis_summary.delivered
+        order_summary.running = analysis_summary.running
+        order_summary.cancelled = analysis_summary.cancelled
+        order_summary.failed = analysis_summary.failed
+
+
+def add_case_summaries(
+    order_summaries: list[OrderSummary],
+    case_summaries: list[CaseSummary],
+) -> None:
+    order_summary_map = {summary.order_id: summary for summary in order_summaries}
+    for case_summary in case_summaries:
+        order_summary = order_summary_map[case_summary.order_id]
+        order_summary.in_sequencing = case_summary.in_sequencing
+        order_summary.in_preparation = case_summary.in_lab_preparation
+
+
+def get_total_cases_in_order(order: Order) -> int:
+    return len(order.cases)
+
+
+def create_case_status_summaries(orders: list[Order]) -> list[CaseSummary]:
+    summaries: list[CaseSummary] = []
+    for order in orders:
+        summary: CaseSummary = get_case_status_summary(order)
+        summaries.append(summary)
+    return summaries
+
+
+def get_case_status_summary(order: Order) -> CaseSummary:
+    in_sequencing: int = get_cases_with_samples_in_sequencing_count(order)
+    in_preparation: int = get_cases_with_samples_in_preparation_count(order)
```

Comment:
Good point 👍 I think that makes sense. The total count should not be exceeded.

Reply:
I updated the logic so that:
1. A case is in lab preparation if any sample in it still has not been prepared.
2. A case is in sequencing if all samples have been prepared and any sample has not been sequenced yet.

Reply:
Great 👍